html/pfappserver/lib/pfappserver/Form/Field/RE2.pm

```diff
@@ -0,0 +1,73 @@
+package pfappserver::Form::Field::RE2;
+
+=head1 NAME
+
+pfappserver::Form::Field::RE2 - A RE2 regex field
+
+=head1 DESCRIPTION
+
+This field extends the default Text field and checks if the input value is an valid RE2 regex
+
+=cut
+
+use HTML::FormHandler::Moose;
+extends 'HTML::FormHandler::Field::Text';
+
+use pf::util;
+use namespace::autoclean;
+
+
+our $class_messages = {
+    'regex' => 'It must be a valid regex',
+};
+
+sub get_class_messages {
+    my $self = shift;
+    return {
+       %{ $self->next::method },
+       %$class_messages,
+    }
+}
+
+apply
+  (
+   [
+    {
+     check => sub {
+         use re::engine::RE2 -strict => 1;
```

Comment:
You probably know better than me, but I think this should go at the top of the file since when a module is used, then its imported at startup even if its there.

I may be wrong though, but just wondering why this isn't with the other imports

Reply:
That will turn on the RE2 engine in only within that scope to for all of the admin.
However we could do an empty import on the top level to detect if it was installed.

Reply:
It would be so that a compile error fires if that is not installed like you said.

So the empty import would be good

Reply:
I just remembered it will fail no matter where it is.
The builds were breaking because I did not add it as a dependency.